code-experiments/src/c_linear.c

```diff
@@ -217,9 +217,17 @@ static coco_problem_t *c_linear_shuffle(coco_problem_t *problem_c,
   
   /* Pick up a random constraint number other than 1.
    * formula for U[M,N]: 
-   * M + rand() / (RAND_MAX / (N - M + 1) + 1)
+   * random = M + (rand() / (RAND_MAX + 1.0)) * (N - M + 1)
```

Comment:
That is what I said. It is exactly what is needed for this function c_linear_shuffle(). The formulae for doubles were modified in c_linear_cons_bbob_problem_allocate() and c_linear_single_cons_bbob_problem_allocate().